src/alchemlyb/parsing/namd.py

```diff
@@ -25,72 +108,162 @@ def extract_u_nk(fep_file, T):
+                if has_idws is False and l[0] == 'FepE_back:':
+                    has_idws = True
+
+                # New window, get IDWS lambda if any
+                # We keep track of lambdas from the #NEW line and if they disagree with the #Free line
+                # within the same file, then complain. This can happen if truncated fepout files
+                # are presented in the wrong order.
+                if l[0] == '#NEW':
+                    lambda1_at_start, lambda2_at_start = float(l[6]), float(l[8])
+                    lambda_idws_at_start = float(l[10]) if 'LAMBDA_IDWS' in l else None
+                    has_idws = True if lambda_idws_at_start is not None else False
+
+                # this line marks end of window; dump data into dataframe
+                if l[0] == '#Free':
+                    # extract lambda values for finished window
+                    # lambda1 = sampling lambda (row), lambda2 = comparison lambda (col)
+                    lambda1 = float(l[7])
+                    lambda2 = float(l[8])
+
+                    # If the lambdas are not what we thought they would be, raise an exception to ensure the calculation
+                    # fails. This can happen if fepouts where one window spans multiple fepouts are processed out of order
+                    # NB: There is no way to tell if lambda_idws changed because it isn't in the '#Free' line that ends a window
+                    if lambda1_at_start is not None \
+                        and (lambda1, lambda2) != (lambda1_at_start, lambda2_at_start):
+                        logger.error(f"Lambdas changed unexpectedly while processing {fep_file}")
+                        logger.error(f"l1, l2: {lambda1_at_start}, {lambda2_at_start} changed to {lambda1}, {lambda2}")
+                        logger.error(line)
+                        raise ValueError("Inconsistent lambda values within the same window")
+
+                    # As we are at the end of a window, convert last window's work and times values to np arrays
+                    # (with energy unit kT since they were kcal/mol in the fepouts)
+                    win_de_arr = beta * np.asarray(win_de) # dE values
+                    win_ts_arr = np.asarray(win_ts) # timesteps
+
+                    # Infer lambda_idws_at_start if it wasn't explictly included in this fepout.
+                    # If lambdas are in ascending order, choose the one before it
+                    # This can happen if user ran minimize before starting dynamics and didn't do firsttimestep 0,
+                    # because NAMD only emits the '#NEW' line on timestep 0 for some reason
+                    if has_idws and lambda_idws_at_start is None:
+                        l1_idx, l2_idx = all_lambdas.index(lambda1), all_lambdas.index(lambda2)
+                        if l1_idx > 0 and l1_idx < l2_idx: # Ascending lambdas
```

Comment:
This conditional has no `else` clause. Is that ok or should it fail if neither `if` nor `elif` is entered?

Reply:
This handles the special case where there are IDWS energies but no lambda_idws value in the current .fepout file, which can happen when NAMD is interrupted.

Reply:
Can you then just add your explanation as a comment to the code, please?